spec/controllers/admin_posts_controller_spec.rb

```diff
@@ -21,42 +21,69 @@
   end
 
   describe "POST #create" do
-    before { fake_login_admin(create(:admin, roles: ["communications"])) }
-
     let(:base_params) { { title: "AdminPost Title",
                           content: "AdminPost content long enough to pass validation" } }
 
-    context "when admin post is valid" do
-      it "redirects to post with notice" do
-        post :create, params: { admin_post: base_params }
-        it_redirects_to_with_notice(admin_post_path(assigns[:admin_post]), "Admin Post was successfully created.")
-      end
-    end
+    authorized_roles = %w[superadmin board board_assistants_team communications support translation]
+    unauthorized_roles = Admin::VALID_ROLES - authorized_roles + [nil]
 
-    context "when admin post is invalid" do
-      context "with invalid translated post id" do
-        it "renders the new template with error message" do
-          post :create, params: { admin_post: { translated_post_id: 0 } }.merge(base_params)
+    context "when admin does not have correct authorization" do
+      let(:admin) { create(:admin) }
 
-          expect(response).to render_template(:new)
-          expect(assigns[:admin_post].errors.full_messages).to include("Translated post does not exist")
-        end
+      unauthorized_roles.each do |admin_role|
+        role_description = admin_role.nil? ? "no role" : "the role #{admin_role}"
 
-        it "doesn't create new tags" do
-          post :create, params: { admin_post: { translated_post_id: 0, tag_list: "badtag" } }.merge(base_params)
-          expect(AdminPostTag.find_by(name: "badtag")).to be_nil
+        context "with #{role_description}" do
+          it "redirects with error" do
+            admin.update(roles: [admin_role])
```

Comment:
We use `update` in pretty much all the existing places, so rather than introduce inconsistency, I'll 
1. use this pull request to change every instance of `update` in these files, not just the ones on the lines I've touched, and
2. make an issue submit another pull request to change it in all the other specs (it's perfectly doable; in fact I did it before writing this comment, just to make sure it wasn't a fools errand)

Should I change the ones in the step definitions for the Cucumber tests as well?

Reply:
:heart: thank you! An yeah, the Cucumber tests are already not super nice to debug, so one fewer thing that could go subtly wrong there would be good IMO. Not super high priority, maybe both of those would be good first-timer-only stories?

Reply:
Well, like I said, I already have the code written, so I don't think it's very nice to make someone else redo it 😆